ksql-engine/src/main/java/io/confluent/ksql/function/udaf/sum/IntegerSumKudaf.java

```diff
@@ -35,11 +36,16 @@
   }
 
   @Override
-  public Integer aggregate(Integer currentVal, Integer currentAggVal) {
-    if (currentVal == null) {
-      return currentAggVal;
+  public Integer aggregate(Integer currentValue, Integer aggregateValue) {
+    if (currentValue == null) {
+      return aggregateValue;
     }
-    return currentVal + currentAggVal;
+    return currentValue + aggregateValue;
+  }
+
+  @Override
+  public Integer undo(Integer valueToUndo, Integer aggregateValue) {
+    return this.aggregate(-1 * valueToUndo, aggregateValue);
```

Comment:
The `DoubleSumKudaf` just does a plain subtraction. Why is this different? Perhaps add a comment to the code if there is a good reason for the discrepancy?